### ISSUE TITLE
catalogProductTierPriceManagementV1GetListGet call fails badly when querying parent product

### ISSUE THREAD

**Issue by gibbotronic** [magento/magento2#24410]
The call to catalogProductTierPriceManagementV1GetListGet fails when querying a configurable product. It should handle the request gracefully but instead dies.

The offending file is
  ```/app/code/Magento/Catalog/Model/Product/TierPriceManagement.php```
This function is wrong:

```php
 public function getList($sku, $customerGroupId)
 {
   $product = $this->productRepository->get($sku, ['edit_mode' => true]);

   $priceKey = 'website_price';
   $value = $this->config->getValue('catalog/price/scope', \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITE);
   if ($value == 0) {
     $priceKey = 'price';
   }

   $cgi = ($customerGroupId === 'all'
           ? $this->groupManagement->getAllCustomersGroup()->getId()
           : $customerGroupId);

   $prices = [];
   foreach ($product->getData('tier_price') as $price) {
     if ((is_numeric($customerGroupId) && (int) $price['cust_group'] === (int) $customerGroupId)
         || ($customerGroupId === 'all' && $price['all_groups'])
         ) {
       /** @var \Magento\Catalog\Api\Data\ProductTierPriceInterface $tierPrice */
       $tierPrice = $this->priceFactory->create();
       $tierPrice->setValue($price[$priceKey])
       ->setQty($price['price_qty'])
       ->setCustomerGroupId($cgi);
       $prices[] = $tierPrice;
     }
   }
   return $prices;
 }
```

The line
```php
   foreach ($product->getData('tier_price') as $price) {
```
might return a null.

It should be
```php
$prices = $product->getData('tier_price');
if ($prices != null) {
  foreach ($prices as $price) {
```

  If you ask for the product info (even a parent configurable product), it'll return data in JSON including
  ```json
    "tier_prices": [] 
  ```
...which correctly indicates that this (configurable) product has no tier prices.
But if you ask the configurable product DIRECTLY for prices using group-prices/all/tiers (eg rest/V1/products/myConfigurableProduct/group-prices/all/tiers) it will error and die as detailed below.

### Preconditions (*)

1. Magento 2.3.2
2. A configurable product with options
3. A child product of the configurable product mentioned in (2)
*Optional:*
4. A tax group so that you can add tiered prices
5. Tiered price for the child product mentioned in (3), for the group mentioned in (4)

### Steps to reproduce (*)

1. Make a GET request to catalogProductTierPriceManagementV1GetListGet http://t213.vg/rest/default/V1/products/{sku}/group-prices/{customerGroupId}/tiers, eg. http://t213.vg/rest/default/V1/products/parentSKU/group-prices/1/tiers

### Expected result (*)

1. The page should return an empty tiers array, eg.
```json
"tier_prices": []
```
2. OR the page should fail with HTTP 400, or HTTP 405 as technically the request is wrong (tier prices are not valid for configurable products)

### Actual result (*)

1. The page dies horribly with an internal error and generates a log file. The page returns something like:
{"message":"Internal Error. Details are available in Magento log file. Report ID: webapi-5d36f243413e9"}
2. The generated log file contains:
```
main.CRITICAL:
  Report ID: webapi-5d36f243413e9; Message: Warning: Invalid argument supplied for foreach() in /usr/local/var/www/vendor/magento/module-catalog/Model/Product/TierPriceManagement.php on line 185 {"exception":"[object] (Exception(code: 0): Report ID: webapi-5d36f243413e9; Message: Warning: Invalid argument supplied for foreach() in /usr/local/var/www/vendor/magento/module-catalog/Model/Product/TierPriceManagement.php on line 185 at /usr/local/var/www/vendor/magento/framework/Webapi/ErrorProcessor.php:206, Exception(code: 0): Warning: Invalid argument supplied for foreach() in /usr/local/var/www/vendor/magento/module-catalog/Model/Product/TierPriceManagement.php on line 185 at /usr/local/var/www/vendor/magento/framework/App/ErrorHandler.php:61)"} [
   ]
```

**Comment by gibbotronic** [comment]
> @gibbotronic do you confirm that you were able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
>     * [x]  yes
> 
>     * [ ]  no
> 

I confirm that this is reproducable on vanilla 2.3.2

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19802` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @gibbotronic. Thank you for your report. 
The issue has been fixed in magento/magento2#24502 by @ashutoshwebkul in 2.3-develop branch
Related commit(s):
 - [114ac4b39a5818df2f1c91c07fd9b27cc97de7b1](https://github.com/magento/magento2/commit/114ac4b39a5818df2f1c91c07fd9b27cc97de7b1)

The fix will be available with the upcoming 2.3.4 release.
